instrumentation/aws-lambda-1.0/library/src/main/java/io/opentelemetry/instrumentation/awslambda/v1_0/ApiGatewayProxyRequest.java

```diff
@@ -19,20 +20,24 @@
 
 abstract class ApiGatewayProxyRequest {
 
-  private static boolean noHttpPropagationNeeded() {
-    List<String> fields = OpenTelemetry.getGlobalPropagators().getTextMapPropagator().fields();
-    return (fields.isEmpty() || xrayPropagationFieldsOnly(fields));
+  private static final boolean noHttpPropagationNeeded;
+
+  static {
+    Collection<String> fields =
+        OpenTelemetry.getGlobalPropagators().getTextMapPropagator().fields();
+    noHttpPropagationNeeded = fields.isEmpty() || xrayPropagationFieldsOnly(fields);
   }
 
-  private static boolean xrayPropagationFieldsOnly(List<String> fields) {
+  private static boolean xrayPropagationFieldsOnly(Collection<String> fields) {
     // ugly but faster than typical convert-to-set-and-check-contains-only
     return (fields.size() == 1)
-        && (ParentContextExtractor.AWS_TRACE_HEADER_PROPAGATOR_KEY.equalsIgnoreCase(fields.get(0)));
+        && (ParentContextExtractor.AWS_TRACE_HEADER_PROPAGATOR_KEY.equalsIgnoreCase(
+            fields.iterator().next()));
```

Comment:
@jkwatson It pains me, but I'm leaning towards reverting the change from `List` to `Collection` - `List` is just easier to use